semgrep-core/src/parsing/tree_sitter/Parse_javascript_tree_sitter.ml

```diff
@@ -722,20 +793,15 @@ and member_expression (env : env) ((v1, v2, v3) : CST.member_expression) : expr
   ObjAccess (v1, v2, PN v3)
 
 and assignment_pattern (env : env) ((v1, v2, v3) : CST.assignment_pattern) =
-  let v1 =
-    match v1 with
-    | `Choice_choice_get x ->
-        let id = anon_choice_rese_id_9a83200 env x in
-        idexp id
-    | `Choice_obj x -> destructuring_pattern env x
-  in
-  let v2 = token env v2 (* "=" *) in
-  let v3 = expression env v3 in
-  (v1, v2, v3)
+  let parameter = pattern env v1 in
+  let pat = parameter_to_pattern parameter in
```

Comment:
we call pattern to get a parameter? and then we convert this parameter (which looks like a pattern ...) to a pattern again?

Reply:
We should probably have two different CST-to-AST functions, one that converts to a parameter and another that converts the CST directly to a pattern.